CRM/Core/Resources.php

```diff
@@ -804,110 +804,35 @@ public static function isAjaxMode() {
 
   /**
    * Provide a list of available entityRef filters.
-   * @todo: move component filters into their respective components (e.g. CiviEvent)
    *
    * @return array
    */
   public static function getEntityRefFilters() {
     $filters = array();
     $config = CRM_Core_Config::singleton();
 
-    if (in_array('CiviEvent', $config->enableComponents)) {
```

Comment:
Let's add in some caching - Civi::cache()-> get & set is preferred I believe - this could be doing a bit of php iteration to check all DAOs that exist

Reply:
The output of this function gets cached in the browser - it's part of the l10n.js dynamic script.

Reply:
OK - I guess that's fine it it's also never loaded in batch processes